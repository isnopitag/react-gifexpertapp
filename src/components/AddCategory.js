import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handeInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    console.log('Handle Submit', inputValue);
    if (inputValue.trim() === "conejinga") {
      setCategories((c) => ["bunny", ...c]);
      alert("Te quiero mucho mi conejita 🐇");
      setInputValue("");
    } else {
      if (inputValue.trim().length > 2) {
        setCategories((c) => [inputValue, ...c]);
        setInputValue("");
      }
    }
  };
  return (
    <form onSubmit={handleSubmit}>

      <p>{inputValue}</p>
      <input type="text" value={inputValue} onChange={handeInputChange} />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
