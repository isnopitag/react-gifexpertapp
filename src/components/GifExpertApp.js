import React, { useState } from "react";
import AddCategory from "./AddCategory.js";
import { GifGrid } from "./GifGrid.js";

const GifExpertApp = () => {
  const [categories, setCategories] = useState(["evangelion"]);

  return (
    <>
      <h2>GifExpertApp</h2>
      <hr />
      <AddCategory setCategories={setCategories} />
      <ol>
        {categories.map((category) => (
          <GifGrid 
            category={category}
            key={category}
            />
        ))}
      </ol>
    </>
  );
};

export default GifExpertApp;
