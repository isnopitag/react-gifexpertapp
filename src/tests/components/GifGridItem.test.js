import React from "react";
import { GifGridItem } from "../../components/GifGridItem";
import { shallow } from "enzyme";

describe("Pruebas en GifGridItem", () => {
  const title = "title";
  const url = "https://localhost/algo.jpg";

  let wrapper = shallow(<GifGridItem title={title} url={url} />);

  beforeEach(() => {
    wrapper = shallow(<GifGridItem title={title} url={url} />);
  });

  test("debe de mostrar <GifGridItem/>", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de tener un parrafo con el title", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("debe de tener laimagen igual a la url y  alt de las props", () => {
    const img = wrapper.find("img");
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("Debe de tener animate__fadeIn", () => {
    const div = wrapper.find("div");

    const className = div.prop("className");

    expect(className.includes("animate__fadeIn")).toBe(true);
  });
});
