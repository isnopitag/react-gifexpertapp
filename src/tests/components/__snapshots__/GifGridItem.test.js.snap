// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem debe de mostrar <GifGridItem/> 1`] = `
<div
  className="card animate__animated animate__fadeIn animate__delay-1s "
>
  <img
    alt="title"
    src="https://localhost/algo.jpg"
  />
  <p>
    title
  </p>
</div>
`;
